modules/testing/src/views/lite/scenarios/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { Button, Intent } from '@blueprintjs/core'
+import _ from 'lodash'
+import React from 'react'
+import { MdPolymer } from 'react-icons/md'
+
+import style from './style.scss'
+import { AppToaster } from './toaster'
+import ScenarioDetails from './ScenarioDetails'
+import SplashScreen from './SplashScreen'
+
+const WEBCHAT_WIDTH = 400
+const DEV_TOOLS_WIDTH = 450
+
+export const updater = { callback: undefined }
+
+export class ScenarioBuilder extends React.Component<Props, State> {
+  private firstEvent
+  private secondEvent
+
+  state = {
+    visible: false,
+    scenario: undefined,
+    errorMessage: undefined,
+    selectedEventIds: []
+  }
+
+  componentDidMount() {
+    updater.callback = this.handleEventClicked
+
+    this.props.store.view.setLayoutWidth(WEBCHAT_WIDTH)
+    this.props.store.view.setContainerWidth(WEBCHAT_WIDTH)
+    this.props.store.view.addHeaderButton({
+      id: 'toggleBuilder',
+      label: 'Show Scenario Builder',
+      icon: <MdPolymer size={18} />,
+      onClick: this.toggleBuilder
+    })
+  }
+
+  componentWillUnmount() {
+    this.props.store.view.removeHeaderButton('toggleBuilder')
+    this.resetWebchat()
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevState.visible && this.state.visible) {
+      this.clearSelection()
+      this.props.store.setMessageWrapper({ module: 'testing', component: 'Wrapper' })
+    } else if (prevState.visible && !this.state.visible) {
+      this.resetWebchat()
+    }
+
+    if (prevState.selectedEventIds !== this.state.selectedEventIds) {
+      this.props.store.view.setHighlightedMessages(this.state.selectedEventIds)
+    }
+  }
+
+  resetWebchat() {
+    this.clearSelection()
+    this.props.store.setMessageWrapper(undefined)
+    this.props.store.view.setLayoutWidth(WEBCHAT_WIDTH)
+    this.props.store.view.setContainerWidth(WEBCHAT_WIDTH)
+  }
+
+  toggleBuilder = () => {
+    this.props.store.view.setContainerWidth(this.state.visible ? WEBCHAT_WIDTH : WEBCHAT_WIDTH + DEV_TOOLS_WIDTH)
+    this.setState({ visible: !this.state.visible })
+  }
+
+  handleEventClicked = eventId => {
+    if (this.firstEvent && this.secondEvent) {
+      return this.clearSelection()
+    }
+
+    if (!this.firstEvent) {
+      this.firstEvent = eventId
+      return this.setState({ selectedEventIds: [eventId] })
+    }
+
+    if (!this.secondEvent) {
```

Comment:
can you rename secondEvent for lastEvent, this way we understand that the user is selecting a range of events and not 2 distinct events